src/app/my-page/homepage-setting/panel/KeywordSetting.tsx

```diff
@@ -49,53 +57,68 @@ const KeywordAddingField = ({
   const textFieldRef = useRef<HTMLInputElement | null>(null)
   const axiosWithAuth = useAxiosWithAuth()
 
-  const handleKeyPress = async (e: React.KeyboardEvent<HTMLInputElement>) => {
-    if (e.key === 'Enter') {
-      e.preventDefault()
-      const trimmed = inputValue.trim()
-      if (trimmed.length < 2) {
-        setToastMessage({
-          severity: 'error',
-          message:
-            '알림 키워드는 양 끝 공백은 제외 최소 2자 이상이어야 합니다.',
-        })
-        return
-      } else if (keywordList?.find((keyword) => keyword.label === trimmed)) {
+  const validateData = (inputValue: string) => {
+    const trimmed = inputValue.trim()
+    if (trimmed.length < 2) {
+      setToastMessage({
+        severity: 'error',
+        message: '알림 키워드는 양 끝 공백은 제외 최소 2자 이상이어야 합니다.',
+      })
+      return false
+    } else if (keywordList?.find((keyword) => keyword.label === trimmed)) {
+      setToastMessage({
+        severity: 'error',
+        message: '이미 등록된 알림 키워드입니다.',
+      })
+      return false
+    }
+    return true
+  }
+
+  const addKeyword = async (trimmed: string) => {
+    await axiosWithAuth
+      .post(`${process.env.NEXT_PUBLIC_API_URL}/api/v1/alarm/add`, {
+        newKeyword: trimmed,
+      })
+      .then(() => {
         setToastMessage({
-          severity: 'error',
-          message: '이미 등록된 알림 키워드입니다.',
-        })
-        return
-      }
-      await axiosWithAuth
-        .post(`${process.env.NEXT_PUBLIC_API_URL}/api/v1/alarm/add`, {
-          newKeyword: trimmed,
+          severity: 'success',
+          message: `'${trimmed}'를 알림 키워드 목록에 추가하였습니다.`,
         })
-        .then(() => {
+        mutate()
+      })
+      .catch((error) => {
+        if (error.response.status === 400) {
           setToastMessage({
-            severity: 'success',
-            message: `'${trimmed}'를 알림 키워드 목록에 추가하였습니다.`,
+            severity: 'error',
+            message: `${error.response.data.message}`,
           })
-          mutate()
-        })
-        .catch((error) => {
-          if (error.response.status === 400) {
-            setToastMessage({
-              severity: 'error',
-              message: `${error.response.data.message}`,
-            })
-          } else {
-            setToastMessage({
-              severity: 'error',
-              message: `'${trimmed}'를 알림 키워드 목록에 추가하지 못했습니다.`,
-            })
-          }
-        })
+        } else {
+          setToastMessage({
+            severity: 'error',
+            message: `'${trimmed}'를 알림 키워드 목록에 추가하지 못했습니다.`,
+          })
+        }
+      })
+  }
+
+  const handleKeyPress = async (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      e.preventDefault()
+      if (validateData(inputValue) === false) return
+      await addKeyword(inputValue.trim())
```

Comment:
개선 제안사항입니닷..
`handleKeyPress` 함수랑 `handleOnClick` 함수 둘 다에 적용되는건데요.
`validateData` 함수 내부에서와 `addKeyword` 함수 둘다에서 `inputValue.trim()` 값을 사용중인데 `validateData` 는 내부에서 trim을 하고 `addKeyword` 에서는 외부에서 trim을 해줘서 실제로 처리하게 되는 데이터가 뭔지 조금 헷갈리는 것 같아요..!
아래처럼 내부에서 처리하는 데이터를 통일시켜주는것도 좋을 것 같습니다 ㅎㅎ..
```suggestion
      if (validateData(inputValue.trim()) === false) return
      await addKeyword(inputValue.trim())
```

Reply:
확인했습니다! 임시적으로 둘로 분활하다보니 이렇게 된 것 같아요:) 처리해서 다시 리뷰 요청하겠습니다~